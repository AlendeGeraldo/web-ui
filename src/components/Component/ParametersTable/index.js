/* eslint-disable jsx-a11y/anchor-is-valid */
import './style.scss';
import React from 'react';
import PropTypes from 'prop-types';
import { Popconfirm, Table, Tag } from 'antd';

const ParametersTable = (props) => {
  const { parameterList, onDelete } = props;

  const tableColumns = [
    {
      title: 'Parâmetro',
      dataIndex: 'name',
      key: 'name',
      sorter: (a, b) => a.name.localeCompare(b.name),
      ellipsis: true,
    },
    {
      title: 'Tipo',
      dataIndex: 'type',
      key: 'type',
    },
    {
      title: 'Valor padrão',
      dataIndex: 'default',
      key: 'default',
    },
    {
      title: 'Obrigatório',
      dataIndex: 'required',
      key: 'requireda',
      render: (required) => {
        let value;
        let color;
        if (required) {
          value = 'Sim';
          color = 'green';
        } else {
          value = 'Não';
          color = 'volcano';
        }
        return (
          <Tag
            color={color}
            style={{ display: 'flex', justifyContent: 'center' }}
          >
            {value}
          </Tag>
        );
      },
    },
    {
      title: 'Detalhes',
      dataIndex: 'details',
      key: 'details',
    },
    {
      title: '',
      key: 'action',
      width: 100,
      render: (record) => (
        <Popconfirm
          value={record}
          title={`Tem certeza de que deseja excluir o parâmetro ${record.name} ?`}
          onConfirm={() => {
            onDelete(record);
          }}
          okText='Sim'
          cancelText='Não'
          placement='left'
        >
          <a href='#'>Remover</a>
        </Popconfirm>
      ),
    },
  ];

  return (
    <Table
      className='parametersTable'
      rowKey={(record) => record.name}
      dataSource={parameterList}
      columns={tableColumns}
      pagination={{ pageSize: 9 }}
      scroll={{ y: 'calc(100vh - 480px)' }}
    />
  );
};

ParametersTable.propTypes = {
  parameterList: PropTypes.arrayOf(PropTypes.object).isRequired,
  onDelete: PropTypes.func.isRequired,
};

export default ParametersTable;
