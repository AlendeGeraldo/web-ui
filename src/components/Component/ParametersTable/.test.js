import React from 'react';

import { shallow, mount } from 'enzyme'

import { act } from 'react-dom/test-utils';

import { Popconfirm, Table, Tag } from 'antd';

import ParametersTable from '.';

const parameters = [
  {
      name: "param1",
      type: "int",
      required: true,
      default: "1",
      details: "detalhes"
  },
  {
      name: "param2",
      type: "string",
      required: false,
      default: "string",
      details: "detalhes"
  }
];

const onDelete = () => {}

describe('ParametersTable component', () => {

  it('is expected render without crashing', () => {
    shallow(<ParametersTable 
      parameterList={[]}
      onDelete = {onDelete} />);
  });

  it('is expected render html correctly', () => {
    const wrapper = shallow(
    <ParametersTable 
      parameterList={[]}
      onDelete = {onDelete} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('is expected to exist Table', () => {
    const wrapper = mount(
      <ParametersTable 
        parameterList={parameters}
        onDelete = {onDelete} />
    );

    expect(
      wrapper
        .find(Table)
        .exists()
    ).toBeTruthy();
  });
});
