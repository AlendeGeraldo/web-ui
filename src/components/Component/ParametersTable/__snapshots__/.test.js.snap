// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParametersTable component is expected render html correctly 1`] = `
<withStore(Table)
  className="parametersTable"
  columns={
    Array [
      Object {
        "dataIndex": "name",
        "ellipsis": true,
        "key": "name",
        "sorter": [Function],
        "title": "Parâmetro",
      },
      Object {
        "dataIndex": "type",
        "key": "type",
        "title": "Tipo",
      },
      Object {
        "dataIndex": "default",
        "key": "default",
        "title": "Valor padrão",
      },
      Object {
        "dataIndex": "required",
        "key": "requireda",
        "render": [Function],
        "title": "Obrigatório",
      },
      Object {
        "dataIndex": "details",
        "key": "details",
        "title": "Detalhes",
      },
      Object {
        "key": "action",
        "render": [Function],
        "title": "",
        "width": 100,
      },
    ]
  }
  dataSource={Array []}
  pagination={
    Object {
      "pageSize": 9,
    }
  }
  rowKey={[Function]}
  scroll={
    Object {
      "y": "calc(100vh - 480px)",
    }
  }
/>
`;
